pkg/api/pluginproxy/access_token_provider.go

```diff
@@ -197,3 +201,15 @@ var getTokenSource = func(conf *jwt.Config, ctx context.Context) (*oauth2.Token,
 func (provider *accessTokenProvider) getAccessTokenCacheKey() string {
 	return fmt.Sprintf("%v_%v_%v_%v", provider.datasourceId, provider.datasourceVersion, provider.route.Path, provider.route.Method)
 }
+
+func addAuthHeaders(reqHeaders *http.Header, route *plugins.AppPluginRoute, data templateData) error {
+	for _, header := range route.TokenAuth.Headers {
+		interpolated, err := InterpolateString(header.Content, data)
+		if err != nil {
+			return err
+		}
+		reqHeaders.Add(header.Name, interpolated)
```

Comment:
@aknuds1  - I felt like appending was safer than overriding just in case there was ever a scenario where we would want other header values in addition to the user-defined values in the config.